packages/arb-token-bridge-ui/src/components/TransferPanel/TransferPanelSummary.tsx

```diff
@@ -182,7 +190,11 @@ export function useGasSummary(
     amountDebounced,
     token,
     shouldRunGasEstimation,
-    latestNetworksAndSigners
+    latestNetworksAndSigners,
+    arbTokenBridge,
+    l1.provider,
+    l2.provider,
+    walletAddress
```

Comment:
not sure which one it is from this list
but when I am connected to the bridge using a smart contract wallet, chose a token, filled in the amount, input a destination address, the summary panel is refreshing constantly. It doesn't refresh like this on production & when connected to this branch with an EOA wallet

Reply:
Yeah having all of these in the dependency array looks fishy and very easy for something to go wrong

Reply:
Yep, it's the `arbTokenBridge` dependency that is causing the re-run and flicker. I will refactor this. Thanks!

Reply:
Update: I moved the hook away from the large file, to its own file. Also, I updated the dependency array to keep only the minimal required dependencies in it without touching the function logic. I've explained the need for each dependency in the comments, and verified that the gas estimation runs ONLY when required.